File: frontend/src/lib/domain/entity.ts
export class Board {
  constructor(
    readonly board: BoardInfo[][],
    readonly boardImg: string[]
  ) {}
}

export class Koma {
  constructor(
    readonly turn: number,
    readonly x: number,
    readonly y: number,
    readonly size: number,
  ) {}
}

export class BoardInfo {
  size: number;
  turn: number;
  constructor(
    turn: number,
    size: number,
  ) {
    this.size = size;
    this.turn = turn;
  }
}

File: backend/api/domain/entity.go
package domain

type Board struct {
	BoardInfo [][]BoardInfo
}

type Koma struct {
	Turn int
	X int
	Y int
	Size int
}

type BoardInfo struct {
	Size int
	Turn int
}

type GameResponse struct {
	Turn int
	X int
	Y int
	Size int
	IsFinished bool
	IsDraw bool
	Next_Turn int
}

func (b *Board) Init() {
	for _, rows := range b.BoardInfo {
		for _, cols := range rows {
			cols.Size = 0
			cols.Turn = 0
		}
	}
}